### ISSUE TITLE
[Feature Request] custom ext names of 'get route config from dir'

### ISSUE THREAD

**Issue by chiaweilee** [umijs/umi#2749]
Is it possible to extend a custom ext names of 'get route config from dir'?

```js
// https://github.com/umijs/umi/blob/master/packages/umi-build-dev/src/routes/getRouteConfigFromDir.js#L8
const JS_EXTNAMES = ['.js', '.jsx', '.ts', '.tsx', /* '.md' */];
```

I am too lazy to write routes myself, I would like to load markdown as React component via loader.

**Comment by sorrycc** [comment]
How could markdown be a route component? You can try to write a umi plugin via [api.modifyRoutes](https://umijs.org/plugin/develop.html#modifyroutes) to handle md files.

**Comment by chiaweilee** [comment]
> How could markdown file be a route component? You can try to write a umi plugin via [api.modifyRoutes](https://umijs.org/plugin/develop.html#modifyroutes) to handle md files.

It could be a route component, see https://github.com/chiaweilee/umi-plugin-md.
Convert markdown into HTML via markdown-it, and then make it be a component.
It's not graceful yet, patch used..

BTW, why I did this?
I am lazy and I wanna write some dairy with markdown, and I don't wish to write routes myself.

I also did it with vue before, https://github.com/chiaweilee/vue-markdown-docs.
